src/traders-controller.ts

```diff
@@ -66,21 +70,39 @@ export class TradersController {
         if (tradersConfig[traderId].insurance_always_enabled) {
           const insuranceTraderConfig =
             tradersConfig[traderId].insurance_config || {};
+          const insuranceConfig: IInsuranceConfig =
+            this.configServer.getConfig<IInsuranceConfig>(
+              ConfigTypes.INSURANCE
+            );
 
           trader.base.insurance.availability = true;
-          trader.base.insurance.min_payment =
-            insuranceTraderConfig.min_payment || 0;
-          trader.base.insurance.min_return_hour =
-            insuranceTraderConfig.min_return_hour || 1;
-          trader.base.insurance.max_return_hour =
-            insuranceTraderConfig.max_return_hour || 2;
-          trader.base.insurance.max_storage_time =
-            insuranceTraderConfig.max_storage_time || 480;
-
-          trader.base.loyaltyLevels.forEach((payloadLevel) => {
-            payloadLevel.insurance_price_coef =
-              insuranceTraderConfig.insurance_price_coef || 1;
-          });
+          if (insuranceTraderConfig.min_payment !== undefined) {
+            trader.base.insurance.min_payment =
+              insuranceTraderConfig.min_payment;
+          }
+          if (insuranceTraderConfig.min_return_hour !== undefined) {
+            trader.base.insurance.min_return_hour =
+              insuranceTraderConfig.min_return_hour;
+          }
+          if (insuranceTraderConfig.max_return_hour !== undefined) {
+            trader.base.insurance.max_return_hour =
+              insuranceTraderConfig.max_return_hour;
+          }
+          if (insuranceTraderConfig.max_storage_time !== undefined) {
+            trader.base.insurance.max_storage_time =
+              insuranceTraderConfig.max_storage_time;
+          }
+          if (insuranceTraderConfig.return_chance_percent !== undefined) {
+            insuranceConfig.returnChancePercent[traderId] = 
+              insuranceTraderConfig.return_chance_percent;
+          }
+          
+          if (insuranceTraderConfig.insurance_price_coef !== undefined) {
+            trader.base.loyaltyLevels.forEach((payloadLevel) => {
+              payloadLevel.insurance_price_coef =
+                insuranceTraderConfig.insurance_price_coef || 100;
```

Comment:
```suggestion
                insuranceTraderConfig.insurance_price_coef;
```

The `|| 100` is not needed anymore since we checked before that is not undefined (and so if it's 0 then it's fine)

Reply:
Do we need to change the definition in config.ts? It’s defined as `number | undefined`

Reply:
Well, it was showing that error at least in VS code - only reason why I still gave that one a fallback and not the others

Reply:
Yeah it look like Typescript is complaining here becausse the `if` is made of out of the function passed to forEach.

I suggest you to rewrite the code like this: 

```ts
const insurancePriceCoef = insuranceTraderConfig.insurance_price_coef;
if (insurancePriceCoef !== undefined) {
  trader.base.loyaltyLevels.forEach((payloadLevel) => {
    payloadLevel.insurance_price_coef = insurancePriceCoef;
  });
}
```

it should fix the ts issue